database/timeline.go

```diff
@@ -0,0 +1,169 @@
+package database
+
+import (
+	"database/sql"
+	"log"
+	"math"
+	"strings"
+	"time"
+
+	"github.com/aau-network-security/haaukins-store/model"
+)
+
+// calculateCost will return a map which is
+// time and number of running vms in total for
+// given time, it is like a timeSeries
+func calculateCost(db *sql.DB) (map[string]int32, error) {
+
+	sT, err := getEarliestDate(db)
+	if err != nil {
+		log.Fatalf("Error get earliest date %v", err)
+	}
+
+	fT, err := getLastDate(db)
+	if err != nil {
+		log.Fatalf("Error get latest date %v", err)
+	}
+	timeSeries := getDates(sT, fT)
+
+	timeSeriesCount := make(map[string]int32)
+	for _, time := range timeSeries {
+		timeSeriesCount[time.Format(TimeFormat)] = 0
+	}
+
+	events := getEvents(db)
+
+	for _, e := range events {
+		sT, err := time.Parse(time.RFC3339, e.StartedAt)
+		if err != nil {
+			log.Fatalf("Error happened %v ", err)
+		}
+		fT, err := time.Parse(time.RFC3339, e.ExpectedFinishTime)
+		if err != nil {
+			log.Fatalf("Error happened %v ", err)
+		}
+		if isInvalidDate(sT) && isInvalidDate(fT) {
+			log.Fatalf("Invalid date formatting !!! ")
+		}
+		invalidTime, err := time.Parse(TimeFormat, "0001-01-01 00:00:00")
+		if err != nil {
+			log.Fatalf("Error happened %v ", err)
+		}
+		iterateOver := getDates(sT, fT)
+		teamCount := getTeamsCount(db, int(e.Id))
+		available := int(e.Available)
+		for _, f := range iterateOver {
+			timeSeriesCount[f.Format(TimeFormat)] += int32(available) + int32(teamCount)
+		}
+		delete(timeSeriesCount, invalidTime.Format(TimeFormat))
+	}
+
+	return timeSeriesCount, nil
+}
+
+// getEvents will query event table
+// without any condition
+func getEvents(db *sql.DB) []model.Event {
+	r, err := db.Query(QueryEventTable)
+	if err != nil {
+		log.Fatalf("Error on executing query %v", err)
+	}
+	var events []model.Event
+	for r.Next() {
+		event := new(model.Event)
+		err := r.Scan(&event.Id, &event.Tag, &event.Name, &event.Available, &event.Capacity, &event.Status, &event.Frontends,
+			&event.Exercises, &event.StartedAt, &event.ExpectedFinishTime, &event.FinishedAt)
+		if err != nil && !strings.Contains(err.Error(), "Null conversion error ") {
+			log.Fatalf("Error on scanning query %v", err)
+		}
+		events = append(events, *event)
+	}
+
+	return events
+}
```

Comment:
The function `func (s *store) GetEvents(in *pb.GetEventRequest) ([]model.Event, error)` we have in the `store.go` can be used. No need to create a new function. i think we can pass `nil` as argument

Reply:
That cannot be used.